commons/src/main/java/io/aiven/kafka/connect/common/source/task/TaskAssignment.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source.task;
+
+/**
+ * TaskAssignment is a common interface which allows source connectors to determine what method of distributing tasks
+ * amongst source connectors in distributed mode.
+ */
+public interface TaskAssignment {
```

Comment:
TaskAssignment sounds a little ambiguous. Maybe `PartitionAssignment` or idk `PartitionassignmentStrategy` or etc.

Reply:
As we are offering alternative options, to just Partition Assignment I think that wouldn't work as well, however I think TaskAssignmentStrategy might be a little better?

Reply:
I was thinking about avoiding task assignment in the name to not to mix with assignment of connector tasks to workers in connect. I'm not sure however what would be a better name here, how would you call files in S3 in this case? I was thinking about PartitionAssignmentStrategy as in assignment of data partitions to connector tasks.

Reply:
RecordAssignmentStrategy ?
Or 
RecordDistributionStrategy ?

because in S3/Azure/GCS etc we are always dealing with SourceRecords right?

Reply:
Actualy even though its in the source package maybe even 
SourceRecordDistribition/AssignmentStartegy

Reply:
@RyanSkraba Your opinion on naming greatly appreciated.

Reply:
In general this change is about cloud objects assigned to tasks, and not specific to partitions, filename and records.
So may be with objects/tasks ? ObjectTaskAssignment ?

Reply:
`SourceObjectDistributionStrategy` is good.

Reply:
I'm ok with all of the options, if others are ok with that I can also make that update

Reply:
I'd avoid an unqualified `Source` as confusing (w.r.t Source workers, source tasks, source bucket) .  If you want to shorten it, I'd go with `ObjectDistributionStrategy`!

Reply:
I'd prefer ObjectDistributionStrategy

Reply:
I think if there is a preference I'll change it to as Murali has mentioned above and I think we'll in a good place then